source/Locales/ScottLocale/ScottLocalesOfScottDomains.lagda

```diff
@@ -0,0 +1,816 @@
+
+open import MLTT.Spartan hiding (𝟚)
+open import MLTT.List hiding ([_])
+open import Slice.Family
+open import UF.FunExt
+open import UF.Logic
+open import UF.PropTrunc
+open import UF.SubtypeClassifier
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.Powerset-MultiUniverse
+open import UF.Size
+open import UF.Classifiers
+open import UF.Univalence
+open import UF.Embeddings
+open import UF.EquivalenceExamples
+open import MLTT.Negation
+
+module Locales.ScottLocale.ScottLocalesOfScottDomains
+        (pt : propositional-truncations-exist)
+        (fe : Fun-Ext)
+        (sr : Set-Replacement pt)
+        (𝓤  : Universe) where
+
+open import DomainTheory.Basics.Dcpo                         pt fe 𝓤
+ renaming (⟨_⟩ to ⟨_⟩∙)
+ hiding   (is-directed)
+open import DomainTheory.Basics.Pointed                      pt fe 𝓤
+ renaming (⊥ to ⊥d)
+open import DomainTheory.Basics.WayBelow                     pt fe 𝓤
+open import DomainTheory.BasesAndContinuity.Bases            pt fe 𝓤
+open import DomainTheory.BasesAndContinuity.Continuity       pt fe 𝓤
+open import DomainTheory.BasesAndContinuity.CompactBasis     pt fe 𝓤
+open import DomainTheory.BasesAndContinuity.ScottDomain      pt fe 𝓤
+open import Locales.ScottLocale.Definition                   pt fe 𝓤
+open import DomainTheory.Topology.ScottTopology              pt fe 𝓤
+open import DomainTheory.Topology.ScottTopologyProperties    pt fe 𝓤
+open import Locales.Frame                                    pt fe
+ hiding (∅)
+open import Locales.Compactness                              pt fe
+ hiding (is-compact)
+open import Locales.Spectrality.SpectralLocale               pt fe
+open import Locales.SmallBasis pt fe sr
+
+open Locale
+open AllCombinators pt fe
+open PropositionalTruncation pt hiding (_∨_)
+
+\end{code}
+
+The module:
+
+\begin{code}
+
+open import Locales.ScottLocale.ScottLocalesOfAlgebraicDcpos pt fe 𝓤
+
+\end{code}
+
+contains a proof that the Scott locale of any algebraic dcpo is a spectral
+locale.
+
+In this module, we extend this proof by showing that the Scott locale is
+spectral.
+
+\section{Preliminaries}
+
+The following function expresses a list being contained in a given subset.
+
+\begin{code}
+
+_⊆⊆_ : {𝓤 𝓥 : Universe} {X : 𝓤  ̇} → List X → 𝓟 {𝓥} {𝓤} X → 𝓤 ⊔ 𝓥  ̇
+_⊆⊆_ {_} {_} {X} xs U = (x : X) → member x xs → x ∈ U
+
+\end{code}
+
+We define the following predicate that expresses what it means for two elements
+of a DCPO `𝓓` to be “bounded above”.
+
+\begin{code}
+
+bounded-above : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → ⟨ 𝓓 ⟩∙ → ⟨ 𝓓 ⟩∙ → Ω (𝓤 ⁺)
+bounded-above 𝓓 x y = ∥ upper-bound (binary-family 𝓤 x y) ∥Ω
+ where
+  open Joins (λ a b → a ⊑⟨ 𝓓 ⟩ₚ b)
+
+infix 30 bounded-above
+
+syntax bounded-above 𝓓 x y = x ↑[ 𝓓 ] y
+
+\end{code}
+
+For the proof of spectrality, we will also need the following decidability
+assumption.
+
+\begin{code}
+
+decidability-condition : (𝓓 : DCPO {𝓤 ⁺} {𝓤}) → 𝓤 ⁺  ̇
+decidability-condition 𝓓 = (c d : ⟨ 𝓓 ⟩∙) →
```

Comment:
Maybe it's worth pointing out that the condition is (trivially) satisfied in the presence of binary joins of compact elements (which are necessarily compact again).